PurchasesTests/Purchasing/PurchasesTests.swift

```diff
@@ -1972,6 +1972,37 @@ class PurchasesTests: XCTestCase {
             expect(self.storeKitWrapper.finishCalled).toEventually(beTrue())
         }
     }
+    
+    func testPurchasingPackageDoesntThrowPurchaseAlreadyInProgressIfCallbackMakesANewPurchase() {
+        setupPurchases()
+        var receivedError: NSError? = nil
+        var secondCompletionCalled = false
+        self.purchases!.offerings { (newOfferings, _) in
+            let package = newOfferings!["base"]!.monthly!
+            self.purchases!.purchasePackage(package) { _,_,_,_  in
+                self.purchases!.purchasePackage(package) { (tx, info, error, userCancelled) in
+                    receivedError = error as NSError?
+                    secondCompletionCalled = true
```

Comment:
this is the error case - while the completion block is running, it hasn't been removed from the list of completion blocks, and so initiating a new purchase will fail. 
I confirmed that this test fails on `develop`

Reply:
good catch!